File: php-pr/51/php/code-quality-naming/UserProfile.php
<?php

class UserProfile
{
    // {fact rule=code-quality-naming@v1.0 defects=0}
    private $processedData;
    private $validItemCount;
    private $hasValidItems;

    public function processStringArray($inputArray)
    {
        $this->processedData = [];
        $this->validItemCount = 0;
        $this->hasValidItems = false;

        foreach ($inputArray as $inputItem) {
            if (!empty($inputItem)) {
                $this->processedData[] = $inputItem;
                $this->validItemCount++;
            }
        }

        $this->hasValidItems = $this->validItemCount > 0;
    }
    // {/fact}
}

File: scala-pr/54/scala/code-quality-naming/UserProfile.scala
class UserProfile {
  // {fact rule=code-quality-naming@v1.0 defects=0}
  private var processedData: List[String] = List.empty
  private var validItemCount: Int = 0
  private var hasValidItems: Boolean = false

  def processStringArray(inputArray: List[String]): Unit = {
    processedData = List.empty
    validItemCount = 0
    hasValidItems = false

    inputArray.foreach { inputItem =>
      if (inputItem.nonEmpty) {
        processedData = processedData :+ inputItem
        validItemCount += 1
      }
    }

    hasValidItems = validItemCount > 0
  }
  // {/fact}
}